glocaltokens/client.py

```diff
@@ -44,64 +44,74 @@ def __init__(
         """
         Initializes a Device. Can set or google_device or ip and port
         """
-        LOGGER.debug("Initializing new Device instance.")
+        log_prefix = f"[Device - {device_name}]"
+        LOGGER.debug(f"{log_prefix} Initializing new Device instance")
         self.device_name = device_name
         self.local_auth_token = None
         self.ip = ip
         self.port = port
         self.google_device = google_device
         self.hardware = hardware
 
-        LOGGER.debug(
-            'Set self device_name to "{}", local_auth_token to None, '
-            "google_device to {}, and hardware to {}".format(
-                device_name, google_device, hardware
+        # Token and name validations
+        if not self.device_name:
+            LOGGER.error(f"{log_prefix} device_name must be provided")
+            return None
+        if not local_auth_token:
+            LOGGER.error(f"{log_prefix} local_auth_token must be provided")
+            return None
+        if not token_utils.is_local_auth_token(local_auth_token):
+            LOGGER.error(
+                f"{log_prefix} local_auth_token must follow the correct format"
             )
-        )
+            return None
```

Comment:
I don't think `__init__` should actually return `None`. Please use just `return` here.
```suggestion
            return
```